### ISSUE TITLE
[question] set CXXFlags from profile when using conan --install

### ISSUE THREAD

**Issue by Viatorus** [conan-io/conan#7015]
I have a CMake project and a profile which I want to use only for testing.

```
include(default)

[env]
CFLAGS="-fsanitize=address -fno-omit-frame-pointer"
CXXFLAGS="-fsanitize=address -fno-omit-frame-pointer"
```

If I run `conan create . -pr=testing`, the flags for the sanitizer are set. If I run `conan install ..` to debug my project, these flags are missing. How can I set these flags through conan install?

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by jgsogo** [comment]
Hi @Viatorus 

`conan install` can be called with the `--profile` argument to, so you can use any profile other than the default one. Use your same `--profile=testing` and it will be used for your install command: `conan install .. --profile=testing`.

Maybe I'm not understanding you scenario.

**Comment by memsharded** [comment]
Hi!

I think the problem is that env-vars are not passed or digested by default by build systems. Depending on the build system (cmake?) it should provide some explicit mechanism to handle these variables. For example:

```python
def build(self):
     cmake = CMake(self):
     cmake.definitions["MY_FLAGS"] = os.getenv("CXXFLAGS")
     # and maybe the CMakeLists.txt needs to process the variable MY_FLAGS to do the right thing
```

You could try to use CMAKE_CXX_FLAGS, but be aware that this overwrite the Release/Debug CXXFLAGS defined by cmake, so it is a problem. This is why Conan uses CONAN_CXX_FLAGS, and then internally it appends them to CMAKE_XXXX variables.

So in summary: it depends on the build system. There is nothing that will make env-vars to be automatically processed by all build systems, and as Conan wraps the build systems, it might require passing these vars correctly following each build system approaches.

**Comment by Viatorus** [comment]
I see, we would have to extend or add a custom CMake generator. 

Thank you. :)

**Comment by shelper** [comment]
related issue https://github.com/conan-io/conan/issues/4754
